content/docs/guides/converting-gatsby-to-tina.mdx

```diff
@@ -0,0 +1,464 @@
+
+```diff
+  const result = await graphql(`
+    {
+      allMdx(sort: { frontmatter: { date: ASC } }, limit: 1000) {
+        nodes {
+          Id
++         body
+// ...
+```
+
+Using the raw markdown from `body` we can generate a custom node that contains the markdown in a format that TinaCMS can read. First We'll import TinaCMS's markdown parser.
+Add this import to the top of `gatsby-node.js`.
+
+```diff
++ const { parseMDX } = require("@tinacms/dist")
+```
+
+Then inside of `onCreateNode` we'll create a new node containing an Abstract Syntax Tree we can pass into our TinaMarkdown component. We'll also need to serialize the result as a string so that we can return it as a single value from our page query.
+
+> Tina converts raw markdown into Abstract Syntax Trees containing the HTML needed to structure the page at build time.
+
+```diff
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+  if (node.internal.type === `Mdx`) {
+    const value = createFilePath({ node, getNode })
++   createNodeField({
++     name: `tinaMarkdown`,
++     node,
++     value: JSON.stringify(parseMDX(node.body)),
++   })
+    createNodeField({
+      name: `slug`,
+      node,
+      value,
+    })
+  }
+}
+```
+
+There's still a few parameters we need to provide to `parseMDX` before it can do it's job.
+
+- The second argument is a copy of the field we defined in our `config.js` file.
+- The third is a callback function that the parser uses when coming across any image links. We'll use that callback function to update our image urls to point to the image directory we defined in our config file.
+
+```diff
++ const imageCallback = url => {
++   return url.replace(/^\./, "/images")
++ }
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+  if (node.internal.type === `Mdx`) {
+    const value = createFilePath({ node, getNode })
+    createNodeField({
+      name: `tinaMarkdown`,
+      node,
+      value: JSON.stringify(
+        parseMDX(
+          node.body,
++          {
++           type: "rich-text",
++           name: "body",
++           label: "Body",
++           isBody: true,
++           },
++           imageCallback
+        )
+      ),
+    })
+// ...
+```
+
+Now that we've defined our new custom node we should be ready to use it in our graphql query. Open `src/templates/blog-post.js` and add our custom node to the page query.
+
+```
+export const pageQuery = graphql`
+  query BlogPostBySlug(
+    $id: String!
+    $previousPostId: String
+    $nextPostId: String
+  ) {
+    site {
+      siteMetadata {
+        title
+      }
+    }
+    mdx(id: { eq: $id }) {
+      id
++     fields {
++       tinaMarkdown
++     }
+// ...
+`
+```
+
+Now we're ready to parse the formatted markdown into our component. We'll also remove the `"children"` prop which normally contains the HTML from Gatsby's mdx parser.
```

Comment:
```suggestion
Now we're ready to parse the formatted markdown into our component. We'll also remove the `children` prop which normally contains the HTML from Gatsby's mdx parser.
```
Don't need the quotes as well as monospace.

Reply:
Done - rephrased all nested quotes to use monospace